### PR TITLE
Handle incorrect batch file loading

### LINKED ISSUE #28368: Soft crash on loading incorrect batch file
This occurs when loading a mask.txt or data.nxs file as a batch file in ISIS SANS interface. The file type for the mask file field is restricted to .txt which makes it less likely that there will be a mix up. But for the batch file there are no restrictions and trying to load and process the wrong file as a batch file is not handled gracefully.

### Expected behavior
Loading a the wrong file in the field Batch file in the ISIS SANS interface should inform the user of the error and handle it gracefully
### Actual behavior
Currently there is an error reporter! and the errors are unhelpful. They can be ValueError or UnicodeDecodeError
### Steps to reproduce the behavior
Load mask file or .nxs in the Batch file field on ISIS SANS interface
### Platforms affected
found in testing for 5.0 release

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of work.**
- Adds a filter so we only can pick CSV files from batch selector
- Convert `csv.Error` into Python Syntax error so we can catch it, preventing the unknown error dialog appearing

**To test:**
- Open the SANS GUI
- Check that hitting Batch File only shows CSV
- Attempt to load some files that are not CSVs (this might be difficult if the filtering does not allow you to pick others, such as on KDE. In this case you will have to type the name of a file in that folder to work around the filter)
- Check that a warning box appears, not the Mantid unknown exception one

Fixes #28368 .

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/SANS/command_interface/batch_csv_file_parser_test.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
 # SPDX - License - Identifier: GPL - 3.0 +
 import os
+import tempfile
```
Comment (tolu28-coder) [inline scripts/test/SANS/command_interface/batch_csv_file_parser_test.py:8]: Everything looks good , but tempfile is imported and i cant find an occurence where it is used.

**Comment by DavidFair** [comment]
I've fixed a Value Error not being caught if you try a txt file with commas, but no numerical values within

**Comment by DavidFair** [comment]
@tolu28-coder sorry for the delay in getting stuff done, this should be good to review again now

**Review by tolu28-coder — APPROVED** [review]
Everything looks great.

**Pull request merged**